homeassistant/components/homekit_controller/button.py

```diff
@@ -115,3 +123,45 @@ async def async_press(self) -> None:
         key = self.entity_description.key
         val = self.entity_description.write_value
         return await self.async_put_characteristics({key: val})
+
+
+class HomeKitEcobeeClearHoldButton(CharacteristicEntity, ButtonEntity):
+    """Representation of a Button control for Ecobee clear hold request."""
+
+    _attr_name = "Clear Hold"
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity is tracking."""
+        return []
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device if any."""
+        prefix = ""
+        if name := super().name:
+            prefix = name
+        return f"{prefix} {self._attr_name}"
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        key = self._char.type
+
+        for val in (False, True):
```

Comment:
Maybe add a comment that this is (we think?) to clear incorrectly cached state in the ecobee, so the 2nd  put actually takes?

Reply:
added comment